### PR TITLE
Create data layer API and use it in Table component

### PR DESCRIPTION
**Pull request by rozamirandar**

### CODE REVIEW

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/soy/TableTag.java [hunk 1]
```diff
@@ -199,21 +216,30 @@ private void _setItems() {
...
-		catch (PortalException e) {
+		catch (PortalException pe) {
 			if (_log.isErrorEnabled()) {
-				_log.error(e);
+				_log.error(pe, pe);
 			}
 		}
 	}
 
+	private static final int _ITEMS_PER_PAGE_DEFAULT_VALUE = 5;
```
Comment (marco-leo) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/soy/TableTag.java]: @rozamirandar @izaera We should make di configurable from system settings

**Comment by izaera** [comment]
Just started reviewing :)

**Comment by izaera** [comment]
Just started reviewing :)

**Comment by marco-leo** [comment]
@izaera during this time we created in the portal the concept of data provider as well, are both a copy of what we did in commerce, i think is better to standardize to one of it, the framework is called info framework, you can ask @JorgeFerrer for more info about it

**Comment by izaera** [comment]
@marco-leo Yes, we were evaluating the `InfoListProvider` framework but discarded it because it doesn't fit our targets completely and also because we think it's easier to create an adapter between both frameworks. That way, our solution (for clay only) is totally decoupled from Liferay (especially from layouts, user, groups, etc, which were almost mandatory in the `InfoListProvider` solution) but can still be used if you have an `InfoListProvider` around :-) .

BTW, this PR does not represent the solution I'm working on now. I have changed several things.

**Pull request closed without merging**